src/components/TreeGrid/TreeGridDataSelectors.ts

```diff
@@ -98,7 +98,7 @@ const sortData = (treeNode: AugmentedTreeNode, sortColumn: string, sortDirection
     // if the last sort configuration differs from the current, we need to resort the children
     // otherwise, performance gains
     if ((<AugmentedTreeNode>treeNode).$meta.sortRequestId !== rootSortRequestId) {
-        sort(treeNode.children, sortDirection, sortColumn, valueGetterForSort);
+        treeNode.children = sort(treeNode.children, sortDirection, sortColumn, valueGetterForSort);
```

Comment:
before the code deliberately used the same array to sort the nodes to avoid allocating new memory. have you noticed any large memory/cpu differences with this new code?

Reply:
try sorting a  couple of times with 500k nodes with the old code and the new one

Reply:
lodash doesn't allow mutable sort, i'll try writing my own implementation

Reply:
ok, but no need for complications if you find out that there is no memory/CPU pressure from this.

Reply:
ok